src/realm-settings/RealmSettingsSection.tsx

```diff
@@ -272,35 +297,33 @@ export const RealmSettingsSection = () => {
               data-testid="rs-keys-tab"
               aria-label="keys-tab"
             >
-              {realmComponents && (
-                <Tabs
-                  activeKey={activeTab}
-                  onSelect={(_, key) => setActiveTab(key as number)}
+              <Tabs
+                activeKey={activeTab}
+                onSelect={(_, key) => setActiveTab(key as number)}
+              >
+                <Tab
+                  id="keysList"
+                  eventKey={0}
+                  data-testid="rs-keys-list-tab"
+                  aria-label="keys-list-subtab"
+                  title={<TabTitleText>{t("keysList")}</TabTitleText>}
                 >
-                  <Tab
-                    id="keysList"
-                    eventKey={0}
-                    data-testid="rs-keys-list-tab"
-                    aria-label="keys-list-subtab"
-                    title={<TabTitleText>{t("keysList")}</TabTitleText>}
-                  >
-                    <KeysListTab realmComponents={realmComponents} />
-                  </Tab>
-                  <Tab
-                    id="providers"
-                    data-testid="rs-providers-tab"
-                    aria-label="rs-providers-tab"
-                    eventKey={1}
-                    title={<TabTitleText>{t("providers")}</TabTitleText>}
-                  >
-                    <KeysProvidersTab
-                      realmComponents={realmComponents}
-                      keyProviderComponentTypes={kpComponentTypes}
-                      refresh={refresh}
-                    />
-                  </Tab>
-                </Tabs>
-              )}
+                  <KeysListTab realmComponents={realmComponents!} />
```

Comment:
If the aforementioned statement is changed there is no need to use a non-null assertion here.
```suggestion
                  <KeysListTab realmComponents={realmComponents} />
```